test/Compiler/Tests.hs

```diff
@@ -72,27 +75,32 @@ disabledTests =
     -- Issue #2640 (forcing translation for runtime erasure) is still open
   , disable "Compiler/.*/simple/Erasure-Issue2640"
     -----------------------------------------------------------------------------
-    -- The following test case fails (at least at the time of writing)
+    -- The following test cases fail (at least at the time of writing)
     -- for the JS backend.
-  , disable "Compiler/JS/simple/Issue4169-2"
+  , disable "Compiler/JS .*/simple/Issue4169-2"
```

Comment:
Please no spaces in file/directory names.

Reply:
I can avoid the spaces by adding custom `Show` and `Read` instances to the `Compiler` data type.
Do you want this? Which is worse, custom `Show` and `Read` instances or spaces in file names?
Currently the custom instances would be small (four cases for each) but that may change later.

Reply:
I think `Show` and `Read` should be altogether avoided, since `Show`s purpose is to turn data into its _Haskell_ source representation.  Just define two custom functions to map between a `Compiler` value and its pretty-printed representation back and forth.  (Could be a simple lookup table (association list) used both ways.)

Reply:
`Show` is also sensitive to adding field names to a data constructors, like going from
````haskell
  | JS JSOptions
```
to 
```haskell
  | JS { jsOptions :: JSOptions }
```
The generated `Show` function will look different for these, and the printing of the section alternative is rather non-sensical as a directory name.

Reply:
I implemented a better solution.